### DISCUSSION TITLE
Download all information on a repository

### DISCUSSION THREAD

**Discussion by brennanwilkes** [community/community#36208] [Repositories]
Hi, I'm trying to calculate some metrics for a sample of public repos, and my scripts require a large number of API calls. I can account for rate limiting just fine, but the network latency of making hundreds of thousand of API calls is slowing my scripts down from taking days to weeks or even months. Most of my queries take the approach of:

- Give me all of the issues marked bug
- Give me all the linked PRs
- Give me all the commits in said PRs
- Give me detailed information on each commit

Is there a way I can simply download a large file / dataset of all (public) information related to a specific repository, and search through it locally?

Apologies if this is in the wrong category,
Thanks

**Comment by ossanna16** [comment]
Hey there @brennanwilkes and thank you for asking a good question! I'm glad you found our community 🙂
